app/views/dashboards/_table_content.html.erb

```diff
@@ -42,9 +42,20 @@
             <% end %>
           <% end %>
 
-          <td><%= link_to call_glyphicon, "#call-#{patient.primary_phone_display}", data: {toggle: "modal"} %></td>
+      <td>
+        <%= link_to call_glyphicon, "#call-#{patient.primary_phone_display}", class:"call_button",
+        data: {
+          patientname: patient.name,
```

Comment:
This is kind of a cool implementation of this! I am going to keep this in the back of my mind when I run into this problem in the future. That said, I worry that we're basically moving the 100 different modals into 100 different data fields and not actually generating any less code with this solution.

It looks like this is basically overruling what's in `calls/_new_call` with a bit of additional javascript to pop in the modal, and I do think there's some merit to keeping that as the flow -- slightly more restful.

I think a way around this might be using rails UJS -- A nice feature of rails views is that they can fire off javascript/vanilla jquery as an action -- e.g. we could rig up `calls/new` to `respond_to :js`, and have a `calls/new.js.erb` that fired off a jquery event like: `$('.call-modal').html(<%= escape_javascript(render 'calls/_new_call', patient: patient %>);`. See `app/views/notes/create.js.erb` for something similar that we're doing. Then we could rig this link to just fire an ajax `get` at a patient new call path. Some reading on how rails does this: http://guides.rubyonrails.org/working_with_javascript_in_rails.html#server-side-concerns https://coderwall.com/p/kqb3xq/rails-4-how-to-partials-ajax-dead-easy

Your thoughts?